src/pages/scanner.tsx

```diff
@@ -0,0 +1,181 @@
+    ctx.fillRect(
+      Math.max(0, boundingBox.x - padding),
+      Math.max(0, boundingBox.y - padding),
+      boundingBox.width + padding * 2,
+      boundingBox.height + padding * 2,
+    );
+  });
+  ctx.restore();
+};
+
+const ScannerPage: NextPage = () => {
+  const [scannedValue, setScannedValue] = useState<string | null>(null);
+  const [scanStatus, setScanStatus] = useState<"idle" | "success" | "error">(
+    "idle",
+  );
+  const scannerMutation = trpc.scanner.scan.useMutation({
+    onSettled: (data) => {
+      setScanStatus("idle");
+      try {
+        // remove succesfully mutated ids from offline queue
+        const existing = localStorage.getItem("offlineQueue");
+        const ids: string[] = existing ? JSON.parse(existing) : [];
+        const filtered = ids.filter((v) => v !== data?.id);
+        localStorage.setItem("offlineQueue", JSON.stringify(filtered));
+      } catch {
+        // not sure what to do with the json errors
+      }
+    },
+    onError: (error) => {
+      console.error(error);
+      setScanStatus("error");
+    },
+  });
+  useEffect(() => {
+    // On page load, remutate any queued IDs
+    try {
+      const existing = localStorage.getItem("offlineQueue");
+      const ids: string[] = existing ? JSON.parse(existing) : [];
+      if (ids.length > 0) {
+        ids.forEach((queuedId) => {
+          scannerMutation.mutate({ id: queuedId, task: "checkIn" });
+        });
+      }
+    } catch {
+      // figure out what to do with the json errors
+    }
+  }, []);
+
+  useEffect(() => {
+    if (scannedValue) {
+      scannerMutation.mutate({
+        id: scannedValue,
+        task: "checkIn",
+      });
+
+      // add scanned id to offline queue for persistence
+      const key = "offlineQueue";
+      const existing = localStorage.getItem(key);
+      const list: string[] = existing ? JSON.parse(existing) : [];
+      list.push(scannedValue);
+      localStorage.setItem(key, JSON.stringify(list));
+    }
+  }, [scannedValue]);
+
+  return (
+    <>
+      <Head>
+        <title>QR Scanner - Deltahacks 12</title>
+      </Head>
+
+      <Drawer
+        pageTabs={[
+          { pageName: "Dashboard", link: "/dashboard" },
+          { pageName: "Scanner", link: "/scanner" },
+        ]}
+      >
+        <main className="px-7 py-16 sm:px-14 lg:pl-20 2xl:pt-20">
+          <div className="max-w-4xl">
+            <h1 className="text-2xl font-bold mb-8 text-black dark:text-white">
+              QR Code Scanner
+            </h1>
+
+            <div className="rounded-md p-6 dark:bg-neutral-800 border-neutral-300 dark:border-neutral-700 bg-white border">
+              <div className="flex flex-col items-center justify-center">
+                <div
+                  className={clsx(
+                    "relative w-full max-w-[320px] border-4 border-dashed rounded-md p-1",
+                    scanStatus === "success"
+                      ? "border-green-500"
```

Comment:
Can you use an object with truthy/falsy values for this instead of the ternary chain?